pkgs/top-level/all-packages.nix

```diff
@@ -8227,6 +8227,43 @@ with pkgs;
     else
       null;
 
+  # We can choose:
+  ccChooser =
+    platform: pkgs: fallback:
+    let
+      # TODO: use cc when https://github.com/NixOS/nixpkgs/pull/365057 is merged
+      inherit (platform) useLLVM isDarwin isGhcjs;
+      useArocc = platform.useArocc or false;
+      useZig = platform.useZig or false;
+      useiOSPrebuilt = platform.useiOSPrebuilt or false;
+      useAndroidPrebuilt = platform.useAndroidPrebuilt or false;
+    in
+    if useiOSPrebuilt then
+      pkgs.darwin.iosSdkPkgs.clang or fallback
+    else if useAndroidPrebuilt then
+      pkgs."androidndkPkgs_${platform.androidNdkVersion}".clang or fallback
+    else if isGhcjs then
+      # Need to use `throw` so tryEval for splicing works, ugh.  Using
+      # `null` or skipping the attribute would cause an eval failure
+      # `tryEval` wouldn't catch, wrecking accessing previous stages
+      # when there is a C compiler and everything should be fine.
+      # NOTE: maybe we remove this and return null instead?
+      throw "no C compiler provided for this platform"
+    else if isDarwin then
+      pkgs.llvmPackages.libcxxClang or fallback
+    else if useArocc then
+      pkgs.arocc or fallback
+    else if useZig then
+      pkgs.zig.cc or fallback
+    else if useLLVM then
+      pkgs.llvmPackages.clang or fallback
+    else
+      # NOTE: we shouldn't imply GCC, ideally we should either fallback, null, or error.
+      pkgs.gcc or fallback;
+
+  # The C compiler to use
```

Comment:
"The C compiler to use" -- under what circumstances exactly? Is it a statement about the present or aspirational?

Do I understand that this takes precedence over what's in the standard stdenv? Does it actually do that on linux? I still see for example this (below), did it get missed?

https://github.com/NixOS/nixpkgs/blob/0ef29681a124ede2135ad259726a3a9f28737b0d/pkgs/stdenv/linux/default.nix#L753

Reply:
We will be dropping CC from the stdenv in the future so this is one of those steps. All future use of `stdenv.cc` should be replaced with `cc`. It is a part of the stdenv and CC wrapper redesign which has been discussed on the stdenv team Matrix.

Reply:
Ok, so what should be changed then?

Reply:
I suggest either removing the comments (I don't think they're adding much as it is?) or elaborate them so that they precisely reflect the intent of the behaviour as it is currently on the master branch once this lands.

You could even include a reference to the design doc so that the intended development trajectory is clear - that would also lead to it being discoverable by those reading the sources, leading to increased collaboration/support/testing on the project.

Reply:
I've updated the comment.